conda_smithy/templates/circle.yml.tmpl

```diff
@@ -24,14 +24,31 @@ dependencies:
   override:
     - docker pull condaforge/linux-anvil
 
+{%- if matrix %}
+{%- if 'ccache-toolchain' in ' '.join(package['meta']['requirements']['build']) %}
+    # Run, test and (if we have a BINSTAR_TOKEN) upload the distributions.
+    - ./ci_support/run_docker_build.sh || touch ./build_failed
```

Comment:
A comment here justifying this would be helpful too: "If any command present on dependencies section fails, CircleCI doesn't cache anything. That's why we need to check for failure later"